include/seqan3/alphabet/nucleotide/dna5.hpp

```diff
@@ -69,174 +70,72 @@ namespace seqan3
-    constexpr dna5 & assign_rank(rank_type const c)
-    {
-        assert(c < value_size);
-        _value = static_cast<internal_type>(c);
-        return *this;
-    }
-    //!\}
-
-    //!\copydoc seqan3::dna4::value_size
-    static constexpr rank_type value_size{5};
+private:
+    //!\brief The base class.
+    using base_t = nucleotide_base<dna5, 5>;
+
+    //!\brief Befriend seqan3::nucleotide_base.
+    friend base_t;
+    //!\cond \brief Befriend seqan3::alphabet_base.
+    friend base_t::base_t;
+    //!\endcond
+    //!\brief Befriend seqan3::rna5 so it can copy #char_to_rank.
+    friend rna5;
 
-    /*!\name Conversion operators
+public:
+    /*!\name Constructors, destructor and assignment
      * \{
      */
-    //!\brief Implicit conversion between dna* and rna* of the same size.
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept and have the same \link value_size \endlink.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type> && value_size == alphabet_size_v<other_nucl_type>
-    //!\endcond
-    constexpr operator other_nucl_type() const noexcept
-    {
-        return other_nucl_type{_value};
-    }
-
-    //!\brief Explicit conversion to any other nucleotide alphabet (via char representation).
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type>
-    //!\endcond
-    explicit constexpr operator other_nucl_type() const noexcept
-    {
-        return detail::convert_through_char_representation<other_nucl_type, std::decay_t<decltype(*this)>>[to_rank()];
-    }
-    //!\}
-
-    //!\name Comparison operators
-    //!\{
-    constexpr bool operator==(dna5 const & rhs) const noexcept
-    {
-        return _value == rhs._value;
-    }
-
-    constexpr bool operator!=(dna5 const & rhs) const noexcept
-    {
-        return _value != rhs._value;
-    }
-
-    constexpr bool operator<(dna5 const & rhs) const noexcept
-    {
-        return _value < rhs._value;
-    }
-
-    constexpr bool operator>(dna5 const & rhs) const noexcept
-    {
-        return _value > rhs._value;
-    }
-
-    constexpr bool operator<=(dna5 const & rhs) const noexcept
-    {
-        return _value <= rhs._value;
-    }
-
-    constexpr bool operator>=(dna5 const & rhs) const noexcept
-    {
-        return _value >= rhs._value;
+    constexpr dna5() : base_t{} {}
+    constexpr dna5(dna5 const &) = default;
+    constexpr dna5(dna5 &&) = default;
+    constexpr dna5 & operator=(dna5 const &) = default;
+    constexpr dna5 & operator=(dna5 &&) = default;
+
+    //!\brief Allow implicit construction from dna/rna of the same size.
+    template <std::Same<rna5> t>    // Accept incomplete type
+    constexpr dna5(t const & r)
+    {
+        using seqan3::to_rank;
+        assign_rank(to_rank(r));
```

Comment:
for`dna15` you use the member function to_rank instead which seems sufficient to me since we know the type. I would change this to be consistent :nail_care: